src/lucky_record/form.cr

```diff
@@ -167,6 +167,18 @@ abstract class LuckyRecord::Form(T)
     end
   end
 
+  def save! : T
+    if save
+      record.not_nil!
+    else
+      raise LuckyRecord::InvalidFormError.new(form_name: typeof(self).to_s, form_object: record)
```

Comment:
Instead of passing record for the form object, I think it should be `self` so that it returns the form object. And I think there is a `form_name` method you can use so you don't need to do `typeof`

Reply:
Actually you're right. typeof(self).to_s is probably better because `form_name` will return the underscored version. So that can probably be kept as-is 👍